input.lisp

```diff
@@ -379,12 +384,12 @@ match with an element of the completions."
          (prompt-lines (ppcre:split #\Newline prompt))
          (prompt-lines-length (length prompt-lines))
          (input-line (input-line-string input))
-         (completions (remove-if (lambda (str)
-                                   (or (string= str "")
-                                       (< (length str) (length input-line))
-                                       (string/= input-line
-                                                 (subseq str 0 (length input-line)))))
-                                 *input-completions*))
+         (completions (take *maximum-completions* (remove-if (lambda (str)
+                                                               (or (string= str "")
+                                                                   (< (length str) (length input-line))
+                                                                   (string/= input-line
+                                                                             (subseq str 0 (length input-line)))))
+                                                             *input-completions*)))
```

Comment:
There should also probably be a deduplication step by wrapping the whole thing in (remove-duplicates).